### ISSUE TITLE
Elastic tabstops

### ISSUE THREAD

**Issue by sphh** [spyder-ide/spyder#2662]
Hi,

Would it be possible to add support for elastic tabstops to spyder?

http://nickgravgaard.com/elastic-tabstops/

Thanks,
Stephan

**Comment by goanpeca** [comment]
Hi @sphh, interesting suggestion, but I am afraid I am :-1: on this as it would actually involve mixing space and tabs in a file..

**Comment by sphh** [comment]
Hi @goanpeca,

I can see your point...

What if we name the idea "elastic whitespace"? It could work as follows:
1. An inline comment is added after the statement (by pressing the tab key and `#` to separate it).
2. When pressing the tab key In the following (or previous) line, the cursor will be aligned with the `#` character.
3. Whenever any of the comments in this particular block moves, all other comments are adjusted accordingly.

With this approach no tab character is needed. It can even be expanded by considering two or more spaces anywhere in any line, but not at the beginning, to be elastic whitespaces. In this case it would be even possible to align tables, comments, ...

What do you think?

**Comment by goanpeca** [comment]
Ok, that could be done I guess... 

@blink1073 thoughts?

*(Not a priority right now... so if you have some time we can point you in the right direction ;-) )*

**Comment by blink1073** [comment]
I think it would be jarring for a programmer, who is used to writing in monospace fonts and having tabs either be *tabs* or a fixed number of spaces.  I am :-1:.

**Comment by goanpeca** [comment]
@blink1073 I think what he meant is the following...

```pyhon
foo = 123   # Comment
bar = [1, 2, 3, 4, 5] # Comment
```

Would somehow turn into this (automagically when typing) the comment in the second line into

```pyhon
foo = 123                # Comment
bar = [1, 2, 3, 4, 5]  # Comment
```

The same idea would apply to definitions using parentheses etc... updating the variable name would adjust the indentation to match.

And if you changed the name of the variable it would also include any adjacent lines with comment, which I think is actually not a bad idea, now maybe it would be too difficult to implement right and not worth the trouble?

**Comment by sphh** [comment]
Thanks @goanpeca, that's exactly what I meant!

It would make a nice feature which would set Spyder further ahead of other IDEs!

**Comment by blink1073** [comment]
Fair enough, but it sounds like quite a mind-bender to implement properly.

**Comment by goanpeca** [comment]
Indeed!

**Comment by ccordoba12** [comment]
I'm -1 on this too. Too much hassle for such a little gain.

**Comment by Nodd** [comment]
I don't know if it could work as an external plugin.

Also from your link:
> SublimeText's "ElasticTabstops" plugin inserts spaces rather than moving tabstops. There's nothing elastic about those tabstops.

This is exactly what we would implement ! But the author seems to be pedantic on this one: the important thing is to keep the comments lined up visually.

**Comment by sphh** [comment]
As an inventor Nick Gravgaard (who invented Elastic Tabstops) has every right on his side to be pendantic ;-) And he might have a point in an ideal world where every editor implements his idea. But as @goanpeca rightly pointed out it's never a good idea to mix spaces and tabs in python files, so the creative adaption of this idea would be a big step forward to make better readable python code in less time. It does not matter if it is called Elastic Whitespace or Magical Alignment or any other great name you can come up with.

An external plugin is perfectly ok.

I would like to help, but my knowledge about writing GUI and text processing code is rather nil.
